src/sidebar/services/api.js

```diff
@@ -99,75 +99,66 @@ function createAPICall(
-      .then(([links, token]) => {
-        const descriptor = get(links, route);
-        const headers = {
-          'Content-Type': 'application/json',
-          'Hypothesis-Client-Version': '__VERSION__', // replaced by versionify
-        };
-
-        if (token) {
-          headers.Authorization = 'Bearer ' + token;
-        }
-
-        const clientId = getClientId();
-        if (clientId) {
-          headers['X-Client-Id'] = clientId;
-        }
-
-        const { url, unusedParams: queryParams } = replaceURLParams(
-          descriptor.url,
-          params
-        );
-        const apiUrl = new URL(url);
-        apiUrl.search = queryString.stringify(queryParams);
-
-        return fetch(apiUrl.toString(), {
+      let response;
+      try {
+        response = await fetch(apiUrl.toString(), {
           body: data ? JSON.stringify(stripInternalProperties(data)) : null,
           headers,
           method: descriptor.method,
-        }).catch(() => {
-          // Re-throw Fetch errors such that they all "look the same" (different
-          // browsers throw different Errors on Fetch failure). This allows
-          // Fetch failures to be either handled in particular ways higher up
-          // or for them to be ignored in error reporting (see `sentry` config).
-          throw new Error(`Fetch operation failed for URL '${url}'`);
         });
-      })
-      .then(response => {
-        let data;
-
-        const status = response.status;
-        if (status >= 200 && status !== 204 && status < 500) {
-          data = response.json();
-        } else {
-          data = response.text();
-        }
-        return Promise.all([response, data]);
-      })
-      .then(
-        ([response, data]) => {
-          // `fetch` executed the request and the response was successfully parsed.
-          onRequestFinished();
-
-          if (response.status >= 400) {
-            // Translate the API result into an `Error` to follow the convention that
-            // Promises should be rejected with an Error or Error-like object.
-            throw translateResponseToError(response, data);
-          }
-
-          return data;
-        },
-        err => {
-          // `fetch` failed to execute the request, or parsing the response failed.
-          onRequestFinished();
-          throw err;
-        }
-      );
+      } catch (err) {
+        // Re-throw Fetch errors such that they all "look the same" (different
+        // browsers throw different Errors on Fetch failure). This allows
+        // Fetch failures to be either handled in particular ways higher up
+        // or for them to be ignored in error reporting (see `sentry` config).
+        throw new Error(`Fetch operation failed for URL '${url}'`);
+      }
+
+      const status = response.status;
+
+      let responseBody;
+      if (status >= 200 && status !== 204 && status < 500) {
+        responseBody = await response.json();
```

Comment:
This is the `Response.json()` call which I believe is triggering errors due to the response having an empty body. Unfortunately the relevant Sentry reports are missing a useful stack trace.

Reply:
Yes, that's exactly the plan.